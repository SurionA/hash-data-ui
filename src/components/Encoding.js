import React from 'react';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const Encoding = ({ addedColumnName, encoding, dispatch }) => {
  return (
    <Grid container>
      <Grid item xs={12}>
        <Box p={2}>
          <FormControl fullWidth={true}>
            <TextField
              id="column_name"
              label="Added column name"
              value={addedColumnName}
              onChange={(event) =>
                dispatch({ addedColumnName: event.target.value })
              }
            />
          </FormControl>
        </Box>
      </Grid>
      <Grid item xs={12}>
        <Box p={2}>
          <FormControl fullWidth={true}>
            <InputLabel id="encoding-select-label">Choose encoding</InputLabel>
            <Select
              id="encoding-select"
              value={encoding}
              onChange={(event) => dispatch({ encoding: event.target.value })}
            >
              <MenuItem value="md5">MD5</MenuItem>
              <MenuItem value="sha256">SHA256</MenuItem>
            </Select>
          </FormControl>
        </Box>
      </Grid>
    </Grid>
  );
};

export default Encoding;
