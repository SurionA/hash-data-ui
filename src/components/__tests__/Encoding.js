import React from 'react';
import renderer from 'react-test-renderer';
import Encoding from '../Encoding';

test('Encoding is being rendered', () => {
  const component = renderer.create(<Encoding />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
