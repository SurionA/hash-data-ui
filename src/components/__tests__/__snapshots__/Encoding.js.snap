// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encoding is being rendered 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="column_name"
            id="column_name-label"
          >
            Added column name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="column_name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          id="encoding-select-label"
        >
          Choose encoding
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="encoding-select"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="encoding-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
